### PR TITLE
Feature/pin message

### PR DESCRIPTION
**Pull request by SeniorZhai**

### CODE REVIEW

**Review by Tougee — APPROVED** [review]

File: app/src/main/java/one/mixin/android/db/PinMessageDao.kt [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+        m.name AS mediaName, m.media_mime_type AS mediaMimeType, m.media_size AS mediaSize, m.media_width AS mediaWidth, m.media_height AS mediaHeight,
+        m.thumb_image AS thumbImage, m.thumb_url AS thumbUrl, m.media_url AS mediaUrl, m.media_duration AS mediaDuration, m.quote_message_id as quoteId,
+        m.quote_content as quoteContent, m.caption as caption, u1.full_name AS participantFullName, m.action AS actionName, u1.user_id AS participantUserId,
+        s.snapshot_id AS snapshotId, s.type AS snapshotType, s.amount AS snapshotAmount, a.symbol AS assetSymbol, s.asset_id AS assetId,
+        a.icon_url AS assetIcon, st.asset_url AS assetUrl, st.asset_width AS assetWidth, st.asset_height AS assetHeight, st.sticker_id AS stickerId,
+        st.name AS assetName, st.asset_type AS assetType, h.site_name AS siteName, h.site_title AS siteTitle, h.site_description AS siteDescription,
+        h.site_image AS siteImage, m.shared_user_id AS sharedUserId, su.full_name AS sharedUserFullName, su.identity_number AS sharedUserIdentityNumber,
```
Comment (YeungKC) [inline app/src/main/java/one/mixin/android/db/PinMessageDao.kt]: We should no need all field?

File: app/src/main/java/one/mixin/android/ui/home/ConversationListFragment.kt [hunk 2]
```diff
@@ -815,6 +818,56 @@ class ConversationListFragment : LinkFragment() {
...
+                            if (id == conversationItem.senderId) {
+                                getText(R.string.chat_you_start)
+                            } else {
+                                conversationItem.senderFullName
+                            },
+                            when {
+                                pinMessage.isImage() -> getText(R.string.chat_pin_image_message)
+                                pinMessage.isVideo() -> getText(R.string.chat_pin_video_message)
+                                pinMessage.isLive() -> getText(R.string.chat_pin_live_message)
+                                pinMessage.isData() -> getText(R.string.chat_pin_data_message)
```
Comment (YeungKC) [inline app/src/main/java/one/mixin/android/ui/home/ConversationListFragment.kt]: This code snippet also appears in PinMessageHolder, should extract a function?

File: app/src/main/java/one/mixin/android/db/PinMessageDao.kt (hunk 1 above)
Comment (SeniorZhai) [reply app/src/main/java/one/mixin/android/db/PinMessageDao.kt]: As you wish

**Review by YeungKC — APPROVED** [review]

File: app/src/main/java/one/mixin/android/db/PinMessageDao.kt [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+import androidx.room.Query
+import androidx.room.RoomWarnings
+import one.mixin.android.vo.ChatHistoryMessageItem
+import one.mixin.android.vo.PinMessage
+import one.mixin.android.vo.PinMessageItem
+import one.mixin.android.vo.PinMessageMinimal
+
+@Dao
+interface PinMessageDao : BaseDao<PinMessage> {
+    @Query("DELETE FROM pin_messages WHERE message_id IN (:messageIds)")
```
Comment (crossle) [inline app/src/main/java/one/mixin/android/db/PinMessageDao.kt:14]: If have 1000 message ids, is it ok?

Comment (SeniorZhai) [reply app/src/main/java/one/mixin/android/db/PinMessageDao.kt:14]: When sending with [chunked](https://github.com/MixinNetwork/android-app/commit/bd27038f1ad03db1868d876aa661a5dfb825baac#diff-f5357d4d283c8186ebaf4fe717962ecd97d24163abc3b2251b01fc03ea123564R144) 128, when receiving, the server limits the maximum pin or unpin size 128.

**Pull request merged**